.github/workflows/macOS-CMake-toolchain.yml

```diff
@@ -0,0 +1,88 @@
+name: macOS (CMake toolchain)
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: macos-latest
+
+    env:
+      # Hardcoded date selecting the latest toolchain from https://swift.org/download/#snapshots.
+      toolchain-date: 2021-01-04
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set environment variables
+        run: |
+          TOOLCHAIN_DATE=${{ env.toolchain-date }}
+          # TOOLCHAIN_SHORT_DATE: TOOLCHAIN_DATE with hypens removed. Used for toolchain identifier and version.
+          TOOLCHAIN_SHORT_DATE="${TOOLCHAIN_DATE//-/}"
+          TOOLCHAIN_NAME=swift-DEVELOPMENT-SNAPSHOT-${TOOLCHAIN_DATE}-a
+          TF_TOOLCHAIN_NAME=swift-tensorflow-DEVELOPMENT-SNAPSHOT-${TOOLCHAIN_DATE}-a
+          echo "TOOLCHAIN_DATE=${TOOLCHAIN_DATE}" >> $GITHUB_ENV
+          echo "TOOLCHAIN_SHORT_DATE=${TOOLCHAIN_SHORT_DATE}" >> $GITHUB_ENV
+          echo "TOOLCHAIN_NAME=${TOOLCHAIN_NAME}" >> $GITHUB_ENV
+          echo "TF_TOOLCHAIN_NAME=${TF_TOOLCHAIN_NAME}" >> $GITHUB_ENV
+          echo "SWIFT_APIS_BUILD_DIR=/tmp/BinaryCache/tensorflow-swift-apis" >> $GITHUB_ENV
+          echo "TOOLCHAIN_ROOT_LOCATION=/Library/Developer/Toolchains/${TOOLCHAIN_NAME}.xctoolchain" >> $GITHUB_ENV
+          echo "TOOLCHAIN_HOME_LOCATION=$HOME/Library/Developer/Toolchains/${TOOLCHAIN_NAME}.xctoolchain" >> $GITHUB_ENV
+
+      - name: Install Ninja
+        run: |
+          brew install ninja
+
+      - name: Install swift.org toolchain
+        run: |
+          curl -sOL https://swift.org/builds/development/xcode/${TOOLCHAIN_NAME}/${TOOLCHAIN_NAME}-osx.pkg
+          xattr -dr com.apple.quarantine ${TOOLCHAIN_NAME}-osx.pkg
+          installer -pkg ${TOOLCHAIN_NAME}-osx.pkg -target CurrentUserHomeDirectory
+          rm -rf ${TOOLCHAIN_NAME}-osx.pkg
+
+      - name: Install pre-built TensorFlow and X10 libraries
+        run: |
+          curl -sL https://artprodeus21.artifacts.visualstudio.com/A8fd008a0-56bc-482c-ba46-67f9425510be/3133d6ab-80a8-4996-ac4f-03df25cd3224/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2NvbXBuZXJkL3Byb2plY3RJZC8zMTMzZDZhYi04MGE4LTQ5OTYtYWM0Zi0wM2RmMjVjZDMyMjQvYnVpbGRJZC80NTU3NC9hcnRpZmFjdE5hbWUvdGVuc29yZmxvdy1kYXJ3aW4teDY00/content?format=zip -o tensorflow-darwin-x64.zip
+          unzip tensorflow-darwin-x64.zip
+          mv tensorflow-darwin-x64/Library/tensorflow-2.4.0 ~/Library/
+
+      - name: Build tensorflow/swift-apis via CMake
+        run: |
+          set -euxo pipefail
+          mkdir -p $SWIFT_APIS_BUILD_DIR
```

Comment:
This is only needed to work around an old CMake bug, I think that with the current GHA builders it should be unnecessary.